nmostesting/suites/IS0402Test.py

```diff
@@ -2218,177 +2218,13 @@ def do_400_check(self, test, resource_type):
-                        "event_type"
-                    ]
-                    for key in keys_to_remove:
-                        if key in data:
-                            del data[key]
-                if version_minor <= 1:
-                    pass
-                if version_minor == 0:
-                    keys_to_remove = [
-                        "channels",
-                        "clock_name",
-                        "grain_rate"
-                    ]
-                    for key in keys_to_remove:
-                        if key in data:
-                            del data[key]
-                return data
-
-            elif resource_type == "flow":
-                if version_minor <= 2:
-                    keys_to_remove = [
-                        "event_type"
-                    ]
-                    for key in keys_to_remove:
-                        if key in data:
-                            del data[key]
-                if version_minor <= 1:
-                    pass
-                if version_minor == 0:
-                    keys_to_remove = [
-                        "bit_depth",
-                        "colorspace",
-                        "components",
-                        "device_id",
-                        "DID_SDID",
-                        "frame_height",
-                        "frame_width",
-                        "grain_rate",
-                        "interlace_mode",
-                        "media_type",
-                        "sample_rate",
-                        "transfer_characteristic"
-                    ]
-                    for key in keys_to_remove:
-                        if key in data:
-                            del data[key]
-                return data
-
-            elif resource_type == "subscription":
-                if version_minor <= 2:
-                    keys_to_remove = [
-                        "authorization"
-                    ]
-                    for key in keys_to_remove:
-                        if key in data:
-                            del data[key]
-                if version_minor <= 1:
-                    pass
-                if version_minor == 0:
-                    keys_to_remove = [
-                        "secure"
-                    ]
-                    for key in keys_to_remove:
-                        if key in data:
-                            del data[key]
-                return data
-
-        # Invalid request
-        return None
-
     def copy_resource(self, type, api_ver=None):
         """Make a clone of the test data for the requested type and API version"""
         if api_ver is None:
             api_ver = self.apis[REG_API_KEY]["version"]
         data = deepcopy(self.test_data[type])
         if self.is04_reg_utils.compare_api_version(api_ver, "v1.3") < 0:
-            data = self.downgrade_resource(type, data, api_ver)
+            data = IS04Utils.downgrade_resource(type, data, api_ver)
```

Comment:
I'm not sure I care, but since `downgrade_resource` now does its own `deepcopy` there's a double one here. 🤷‍♂️

Reply:
Hmmmmm

Perhaps:

```
if self.is04_reg_utils.compare_api_version(api_ver, "v1.3") < 0:
    data = IS04Utils.downgrade_resource(type, self.test_data[type], api_ver)
else
    data = deepcopy(self.test_data[type])

return data
```